client/src/main/java/org/apache/nemo/client/beam/NemoPipelineResult.java

```diff
@@ -64,15 +64,18 @@ public State cancel() throws IOException {
 
   @Override
   public State waitUntilFinish(final Duration duration) {
-    final boolean finished;
     try {
-      finished = this.jobDone.await(duration.getMillis(), TimeUnit.MILLISECONDS);
-      if (finished) {
-        LOG.info("Job successfully finished before timeout of {}ms, while waiting until finish", duration);
-      } else {
-        LOG.info("Job timed out before {}ms, while waiting until finish.", duration);
-        JobLauncher.shutdown();
+      if (duration.getMillis() < 1) {
         this.jobDone.await();
+      } else {
+        final boolean finished = this.jobDone.await(duration.getMillis(), TimeUnit.MILLISECONDS);
+        if (finished) {
+          LOG.info("Job successfully finished before timeout of {}ms, while waiting until finish", duration);
+        } else {
+          LOG.info("Job timed out before {}ms, while waiting until finish.", duration);
+          JobLauncher.shutdown();
```

Comment:
If the job is not finished until the duration, I think it should just return the current state of the job, 
instead of shutting down the job here.  
Users can decide to wait more or shut down right now by calling `PipelineResult.cancel()`. 
@wonook What do you think?

Reply:
The API says `   * @return The final state of the pipeline or null on timeout.`